### PR TITLE
Add operation weights

### LINKED ISSUE #816: Add operation weights

### PR DESCRIPTION
**Pull request by gnawin**
# Pull request details

## Describe the changes made in this pull request

- Add a function to calculate the intervals of the milestone years
- Add weights to operation costs terms

## List of related issues or pull requests

Closes #816 

## Collaboration confirmation

As a contributor I confirm

-   [x] I read and followed the instructions in README.dev.md
-   [ ] The documentation is up to date with the changes introduced in this Pull Request (or NA)
-   [x] Tests are passing
-   [x] Lint is passing

### CODE REVIEW

File: src/create-model.jl [hunk 1]
```diff
@@ -525,6 +525,24 @@ function profile_aggregation(agg, profiles, year, commission_year, key, block, d
...
+
+Create a dictionary of intervals for `years`. The interval is assigned to the its starting year.
+The last interval is 1.
+"""
+function create_intervals_for_years(years)
+    intervals = Dict()
+
+    for i in 1:length(years)-1
+        intervals[years[i]] = years[i+1] - years[i]
+    end
```
Comment (abelsiqueira) [inline src/create-model.jl:539]:
```suggestion
    # This assumes that `years` is ordered
    for i in 1:length(years)-1
        intervals[years[i]] = years[i+1] - years[i]
    end
```

**Review by abelsiqueira — APPROVED** [review]
Thanks for the PR. There is currently no guarantee that `years` is ordered, as far as I remember. The TulipaIO reading follows the order of the file, so it should be good to ensure that somehow. The earliest place would in `io.jl`, changing the `get_table` to a `SELECT * FROM year_data ORDER BY year` or similar. It's otherwise good to go, so I'll pre-approve but feel free to ask for another review.

**Comment by gnawin** [comment]
I'll check the tests

**Comment by gnawin** [comment]
@abelsiqueira Using get_table directly did not work :(, so I changed to `DBInterface.execuete()` following examples for other data. Does this look fine?

**Review by abelsiqueira — APPROVED** [review]
It looks correct. If the tests pass, it's approved.

**Pull request merged**